app/services/onlinetesting/Phase3TestService.scala

```diff
@@ -0,0 +1,196 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.onlinetesting
+
+import _root_.services.AuditService
+import config.LaunchpadGatewayConfig
+import connectors.launchpadgateway.exchangeobjects._
+import connectors._
+import connectors.launchpadgateway.LaunchpadGatewayClient
+import factories.{ DateTimeFactory, UUIDFactory }
+import model.OnlineTestCommands._
+import model.ProgressStatuses
+import model.persisted.phase3tests.{ Phase3Test, Phase3TestGroup }
+import org.joda.time.DateTime
+import repositories.application.GeneralApplicationRepository
+import repositories._
+import repositories.onlinetesting.Phase3TestRepository
+import services.events.{ EventService, EventSink }
+import services.onlinetesting.Phase3TestService.InviteLinkCouldNotBeCreatedSuccessfully
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+import scala.language.postfixOps
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object Phase3TestService extends Phase3TestService {
+  import config.MicroserviceAppConfig._
+  val appRepository = applicationRepository
+  val p3TestRepository = phase3TestRepository
+  val cdRepository = faststreamContactDetailsRepository
+  val launchpadGatewayClient = LaunchpadGatewayClient
+  val tokenFactory = UUIDFactory
+  val dateTimeFactory = DateTimeFactory
+  val emailClient = Phase3OnlineTestEmailClient
+  val auditService = AuditService
+  val gatewayConfig = launchpadGatewayConfig
+  val eventService = EventService
+
+  case class InviteLinkCouldNotBeCreatedSuccessfully(message: String) extends Exception(message)
+}
+
+trait Phase3TestService extends OnlineTestService with ResetPhase3Test with EventSink {
+  val appRepository: GeneralApplicationRepository
+  val p3TestRepository: Phase3TestRepository
+  val cdRepository: contactdetails.ContactDetailsRepository
+  val launchpadGatewayClient: LaunchpadGatewayClient
+  val tokenFactory: UUIDFactory
+  val dateTimeFactory: DateTimeFactory
+  val emailClient: OnlineTestEmailClient
+  val auditService: AuditService
+  val gatewayConfig: LaunchpadGatewayConfig
+
+  override def nextApplicationReadyForOnlineTesting: Future[List[OnlineTestApplication]] = {
+    p3TestRepository.nextApplicationsReadyForOnlineTesting
+  }
+
+  override def registerAndInviteForTestGroup(
+  application: List[OnlineTestApplication])(implicit hc: HeaderCarrier): Future[Unit] = Future.failed(new NotImplementedError())
```

Comment:
It's better to use: ??? for something which is not implemented. It's more visible